clang/include/clang/Basic/CodeGenOptions.h

```diff
@@ -404,6 +404,15 @@ class CodeGenOptions : public CodeGenOptionsBase {
   /// List of pass builder callbacks.
   std::vector<std::function<void(llvm::PassBuilder &)>> PassBuilderCallbacks;
 
+  /// List of global variables explicitly specified by the user as toc-data.
+  std::vector<std::string> TocDataVarsUserSpecified;
+
+  /// List of global variables that over-ride the toc-data default.
+  std::vector<std::string> NoTocDataVars;
+
+  /// Flag for all global variables to be treated as toc-data.
+  bool AllTocData;
```

Comment:
The uninitialized `AllTocData` would cause memorysanitizer failures.
booleans and enums should be moved to `CodeGenOptionsBase` using `clang/Basic/CodeGenOptions.def`.
I fixed this in 605abe0689dfd28aadc9413306f33a4494cf3fb8

Reply:
Thank you!